src/components/Input/Input.js

```diff
@@ -162,13 +169,33 @@ const InputElement = styled(HtmlElement)`
   ${inputSuffixStyles};
 `;
 
+const InputTooltip = styled(Tooltip)`
+  ${tooltipBaseStyles};
+`;
+
+const determineSuffix = ({ invalid, hasWarning, showValid }) => {
+  /* eslint-disable react/prop-types */
```

Comment:
Hm, yeah, I get that using a switch here was a bit difficult/unnecessary. Looking at the code a bit more, I came up with this solution. I find it a bit more elegant, but it probably has worse runtime implications. Not worth sweating about. Just putting this here because I spent some time thinking. 😉 

```javascript
const getSuffixComponent = ({ invalid, hasWarning, showValid }) => {
  const components = [
    hasWarning && WarningIcon,
    hasWarning && ErrorIcon,
    showValid && ValidIcon
  ];

  const SuffixComponent = find(identity, components);

  return ({ className }) => SuffixComponent ? <SuffixComponent className={className} /> : null;
};
```